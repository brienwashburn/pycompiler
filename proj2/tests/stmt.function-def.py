def _1(): pass
def _2(a): pass
def _3(a : int): pass
def _4(a,): pass
def _5(a : int = b): pass
def _6(a : int = b,): pass
def _7(a : int = b, q : bar = c): pass
def _8(a = b, q = c): pass
def _9(q, r, x : str, a : foo = b, l = m, n = o): pass
def _10(q, r, x : str, a : foo = b, l = m, n = o,): pass
def _11(a, *, l): pass
def _12(a, *b): pass
def _13(a, *b : int): pass
def _14(a = quux, *b : int): pass
def _15(a, *, foo = quux): pass
def _16(a, *, foo : int = quux): pass
def _17(a, *rax : baz, foo : int = quux): pass
def _18(a, *rax : baz, foo : int = quux, d, e = f, g : str): pass
def _19(a, *rax : baz, foo : int = quux, d, e = f, g : str, **blah): pass
def _20(a, *rax : baz, foo : int = quux, d, e = f, g : str, **blah : int): pass
def _21(a, *baz : quux **what): pass
def _22(a, **bar): pass
def _23(**bar): pass
def _24(*, a): pass
def _25(*, a : int = b): pass
def _26(*, a : int = b, **foo): pass
def _27(a, d : int, b = c, e : int = f, *g : int, h = i, j : int = k, l, m : int, **n : int): pass

def _28(a, d : int, b = c, e : int = f, *g : int, h = i, j : int = k, l, m : int, **n : int):
    a = 1
    b = 2
    c = 3
    a, b, c = foo(lambda x: y, z) + q

@_29d
def _29(): pass

@_30d.a.b.c
def _30(): pass

@_31d(a)
def _31(): pass

@_32d(a, d, b = c, e = f, *g, h = i, j= k, **n)
def _32(): pass

@_33d.a.b.c(a, d, b = c, e = f, *g, h = i, j= k, **n)
def _33(): pass

def _34() -> int: pass

@_35d1
@_35d2.a.b.c
@_35d3(a)
@_35d4.a.b.c(a, d, b = c, e = f, *g, h = i, j= k, **n)
def _35(a, d : int, b = c, e : int = f, *g : int, h = i, j : int = k, l, m : int, **n : int) -> int:
    a = 1
    b = 2
    c = 3
    a, b, c = foo(lambda x: y, z) + q
