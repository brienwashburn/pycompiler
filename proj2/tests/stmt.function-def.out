(Module
 (FunctionDef
  (name _1)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _2)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _3)
  (args
   (Arguments
    (args a)
    (arg-types (Name int))
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _4)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _5)
  (args
   (Arguments
    (args a)
    (arg-types (Name int))
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults (Name b))))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _6)
  (args
   (Arguments
    (args a)
    (arg-types (Name int))
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults (Name b))))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _7)
  (args
   (Arguments
    (args a q)
    (arg-types (Name int) (Name bar))
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults (Name b) (Name c))))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _8)
  (args
   (Arguments
    (args a q)
    (arg-types #f #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults (Name b) (Name c))))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _9)
  (args
   (Arguments
    (args q r x a l n)
    (arg-types #f #f (Name str) (Name foo) #f #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f #f #f (Name b) (Name m) (Name o))))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _10)
  (args
   (Arguments
    (args q r x a l n)
    (arg-types #f #f (Name str) (Name foo) #f #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f #f #f (Name b) (Name m) (Name o))))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _11)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs l)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _12)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg b)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _13)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg b (Name int))
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _14)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg b (Name int))
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults (Name quux))))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _15)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs foo)
    (kwonlyarg-types #f)
    (kw_defaults (Name quux))
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _16)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs foo)
    (kwonlyarg-types (Name int))
    (kw_defaults (Name quux))
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _17)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg rax (Name baz))
    (kwonlyargs foo)
    (kwonlyarg-types (Name int))
    (kw_defaults (Name quux))
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _18)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg rax (Name baz))
    (kwonlyargs foo d e g)
    (kwonlyarg-types (Name int) #f #f (Name str))
    (kw_defaults (Name quux) #f (Name f) #f)
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _19)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg rax (Name baz))
    (kwonlyargs foo d e g)
    (kwonlyarg-types (Name int) #f #f (Name str))
    (kw_defaults (Name quux) #f (Name f) #f)
    (kwarg blah)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _20)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg rax (Name baz))
    (kwonlyargs foo d e g)
    (kwonlyarg-types (Name int) #f #f (Name str))
    (kw_defaults (Name quux) #f (Name f) #f)
    (kwarg blah (Name int))
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _21)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg baz (BinOp (Name quux) Pow (Name what)))
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _22)
  (args
   (Arguments
    (args a)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg bar)
    (defaults #f)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _23)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg bar)
    (defaults)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _24)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _25)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types (Name int))
    (kw_defaults (Name b))
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _26)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types (Name int))
    (kw_defaults (Name b))
    (kwarg foo)
    (defaults)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _27)
  (args
   (Arguments
    (args a d b e)
    (arg-types #f (Name int) #f (Name int))
    (vararg g (Name int))
    (kwonlyargs h j l m)
    (kwonlyarg-types #f (Name int) #f (Name int))
    (kw_defaults (Name i) (Name k) #f #f)
    (kwarg n (Name int))
    (defaults #f #f (Name c) (Name f))))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _28)
  (args
   (Arguments
    (args a d b e)
    (arg-types #f (Name int) #f (Name int))
    (vararg g (Name int))
    (kwonlyargs h j l m)
    (kwonlyarg-types #f (Name int) #f (Name int))
    (kw_defaults (Name i) (Name k) #f #f)
    (kwarg n (Name int))
    (defaults #f #f (Name c) (Name f))))
  (body
   (Assign (targets (Name a)) (value (Num 1)))
   (Assign (targets (Name b)) (value (Num 2)))
   (Assign (targets (Name c)) (value (Num 3)))
   (Assign
    (targets (Tuple (Name a) (Name b) (Name c)))
    (value
     (BinOp
      (Call
       (func (Name foo))
       (args
        (Lambda
         (Arguments
          (args x)
          (arg-types #f)
          (vararg #f)
          (kwonlyargs)
          (kwonlyarg-types)
          (kw_defaults)
          (kwarg #f)
          (defaults #f))
         (Name y))
        (Name z))
       (keywords)
       (starargs #f)
       (kwargs #f))
      Add
      (Name q)))))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _29)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list (Name _29d))
  (returns #f))
 (FunctionDef
  (name _30)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list (Attribute (Attribute (Attribute (Name _30d) a) b) c))
  (returns #f))
 (FunctionDef
  (name _31)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list
   (Call
    (func (Name _31d))
    (args (Name a))
    (keywords)
    (starargs #f)
    (kwargs #f)))
  (returns #f))
 (FunctionDef
  (name _32)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list
   (Call
    (func (Name _32d))
    (args (Name a) (Name d))
    (keywords (b (Name c)) (e (Name f)) (h (Name i)) (j (Name k)))
    (starargs (Name g))
    (kwargs (Name n))))
  (returns #f))
 (FunctionDef
  (name _33)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list
   (Call
    (func (Attribute (Attribute (Attribute (Name _33d) a) b) c))
    (args (Name a) (Name d))
    (keywords (b (Name c)) (e (Name f)) (h (Name i)) (j (Name k)))
    (starargs (Name g))
    (kwargs (Name n))))
  (returns #f))
 (FunctionDef
  (name _34)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list)
  (returns (Name int)))
 (FunctionDef
  (name _35)
  (args
   (Arguments
    (args a d b e)
    (arg-types #f (Name int) #f (Name int))
    (vararg g (Name int))
    (kwonlyargs h j l m)
    (kwonlyarg-types #f (Name int) #f (Name int))
    (kw_defaults (Name i) (Name k) #f #f)
    (kwarg n (Name int))
    (defaults #f #f (Name c) (Name f))))
  (body
   (Assign (targets (Name a)) (value (Num 1)))
   (Assign (targets (Name b)) (value (Num 2)))
   (Assign (targets (Name c)) (value (Num 3)))
   (Assign
    (targets (Tuple (Name a) (Name b) (Name c)))
    (value
     (BinOp
      (Call
       (func (Name foo))
       (args
        (Lambda
         (Arguments
          (args x)
          (arg-types #f)
          (vararg #f)
          (kwonlyargs)
          (kwonlyarg-types)
          (kw_defaults)
          (kwarg #f)
          (defaults #f))
         (Name y))
        (Name z))
       (keywords)
       (starargs #f)
       (kwargs #f))
      Add
      (Name q)))))
  (decorator_list
   (Name _35d1)
   (Attribute (Attribute (Attribute (Name _35d2) a) b) c)
   (Call
    (func (Name _35d3))
    (args (Name a))
    (keywords)
    (starargs #f)
    (kwargs #f))
   (Call
    (func (Attribute (Attribute (Attribute (Name _35d4) a) b) c))
    (args (Name a) (Name d))
    (keywords (b (Name c)) (e (Name f)) (h (Name i)) (j (Name k)))
    (starargs (Name g))
    (kwargs (Name n))))
  (returns (Name int))))
